query/iterator.gen.go.tmpl

```diff
@@ -410,11 +411,75 @@ func (itr *{{$k.name}}SortedMergeIterator) pop() (*{{$k.Name}}Point, error) {
 type {{$k.name}}SortedMergeHeap struct {
 	opt   IteratorOptions
 	items []*{{$k.name}}SortedMergeHeapItem
+	// if each input comes from a unique single time series, we can make a shortcut.
+	// detection of the shortcut introduces some overhead but it gets significant
+	// performance improvement in cases like SELECT * FROM m GROUP BY *
+	fast bool
+}
+
+func (h *{{$k.name}}SortedMergeHeap) detectFast() {
+       for _, item := range h.items {
```

Comment:
Is it possible that `len(h.items) == 1`? If so, then you could add a shortcut here:

```go
if len(h.items) == 1 {
    return
}
```

Reply:
looks it's already handled at line 224:

https://github.com/influxdata/influxdb/blob/a0f2e0c21aada3663f9960f1b773910572f5cc74/query/iterator.go#L220-L226